dftb-uv_2d.py

```diff
@@ -200,49 +178,31 @@ def smooth_spectrum(path, min_energy, max_energy, min_wavelength, max_wavelength
         xmin_spectrum = min(0.0, min_energy)
         xmax_spectrum = math.ceil(max_energy) + spectrum_discretization_step
 
-    # global lists
-    statelist = list()  # mode
-    energylist = list()  # energy eV
-    intenslist = list()  # fosc
     gauss_sum = list()  # list for the sum of single gaussian spectra = the convoluted spectrum
 
-    spectrum_file = path + '/' + 'EXC.DAT'
+    spectrum_file = path + '/' + dir + '/' + 'EXC.DAT'
 
     # open a file
     # check existence
     try:
-        with open(spectrum_file, "r") as input_file:
-            count_line = 0
-            for line in input_file:
-                # only recognize lines that start with number
-                # split line into 3 lists mode, energy, intensities
-                # line should start with a number
-                if 5 <= count_line <= 55:
-                    if re.search("\d\s{1,}\d", line):
-                        energylist.append(float(line.strip().split()[0]))
-                        intenslist.append(float(line.strip().split()[1]))
-                else:
-                    pass
-                count_line = count_line + 1
+        energylist, intenslist = read_dftb_output(spectrum_file, read_dftb_outoput_start_line, read_dftb_outoput_end_line)
 
     # file not found -> exit here
     except IOError:
         print(f"'{spectrum_file}'" + " not found", flush=True)
-        sys.exit(1)
+    except Exception as e:
+        print("Rank: ", comm_rank, " encountered Exception: ", e.message, e.args)
 
```

Comment:
Should the function return if this Exception is hit, or should it proceed as it currently does?

Reply:
If the smoothing of one molecule assigned to an MPI process is not successful, I would still like for the MPI process to continue spanning all the other molecules that it has been assigned. Because of this, I removed the `sys.exit(1)`

Reply:
The problem is that if this Exception is encountered, then `energylist` defined on line 188 could be undefined, but it is referenced on line 209. You need to add code to explicitly handle the situation in which the energylist variable is empty/None/undefined.

Reply:
You are right. 
I added a new try-catch handler for this. 
Let me know if this look fine.